src/localization/features/feature_buffer.cpp

```diff
@@ -46,21 +47,12 @@ void FeatureBuffer::deleteFeature() {
   ids.erase(ids.begin());
 }
 
-void FeatureBuffer::addFeature(int id, const iFeature::ConstPtr &feature) {
-  if (ids.size() > static_cast<size_t>(std::numeric_limits<int>::max())) {
-    fprintf(stderr,
-            "[ERROR] ids vector size does not fit in integer type. Cannot add "
-            "feature with id: %d.\n",
-            id);
-    return;
-  }
+void FeatureBuffer::addFeature(int id,  std::unique_ptr<iFeature> feature) {
```

Comment:
You might want to get an rref here, i.e., 
```cpp
void FeatureBuffer::addFeature(int id,  std::unique_ptr<iFeature>&& feature) {
```
This is what anyway _is_ happening under the hood as the compiler is able to figure out that you want an rref here, so we might as well be explicit about it.